File: TypeScript/1684.ts
/**
 * 1684. Count the Number of Consistent Strings
 * https://leetcode.com/problems/count-the-number-of-consistent-strings/
 */

function countConsistentStrings(allowed: string, words: string[]): number {
  let out: number = 0;

  words.forEach((word: string) => {
    let flag: boolean = false;

    for (let i: number = 0; i < word.length; i++) {
      flag = allowed.indexOf(word[i]) < 0;
      if (flag) break;
    }

    if (!flag) out += 1;
  });

  return out;
}

File: Rust/1684.rs
/**
 * 1684. Count the Number of Consistent Strings
 * https://leetcode.com/problems/count-the-number-of-consistent-strings/
 */

impl Solution {
  pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
    let mut out: i32 = 0;

    for word in words {
      let mut flag: bool = false;

      for w in word.chars() {
        match allowed.find(w) {
          Some(_) => {
            continue;
          }
          None => {
            flag = true;
            break;
          }
        }
      }

      if !flag {
        out += 1;
      }
    }

    return out;
  }
}
